src/Authentication/App/Controllers/LogoutController.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Lightit\Authentication\App\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Lightit\Authentication\Domain\Actions\LogoutAction;
+use Lightit\Models\JWTAuthenticatable;
+use Lightit\Shared\App\Exceptions\Http\UnauthorizedException;
+
+class LogoutController
+{
+    public function __invoke(Request $request, LogoutAction $logoutAction): Response
+    {
+        /** @var JWTAuthenticatable|null $user */
+        $user = $request->user();
+
+        if (! $user) {
+            throw new UnauthorizedException();
+        }
```

Comment:
You can get the authenticated user like this

```suggestion
    public function __invoke(
        #[CurrentUser]
        User $user,
        LogoutAction $logoutAction
    ): Response {
```

You shouldn't check for its existence since the middleware (if it doesn't have the auth middleware, add it) provides that validation